src/index.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+function Modal(clickElm, modalParent, options) {
+  options = options || {};
+
+  var that = this;
+  clickElm.addEventListener('click', function(e) {
+    e.preventDefault();
+    that.show();
+  }, false);
+  this.clickElm = clickElm;
+  this.modalParent = modalParent || document.documentElement;
```

Comment:
this could be a feature we add later but I think since this object would be instantiated multiple times per product "buy" button (right?) that the default should be to attempt to re-attach to the last parent used so that it hides and shows the same modal.